src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -732,7 +786,16 @@ public static unsafe bool TryCompleteReceiveFrom(SafeSocketHandle socket, Span<b
                 Interop.Error errno;
                 int received;
 
-                if (buffers != null)
+                if (!socket.IsSocket)
```

Comment:
How would we end up here when !socket.IsSocket? 

I'm assuming that pipes only need simple send/receive operations. What would cause a ReceiveFrom on a pipe?

Reply:
Similarly below, for SendTo

Reply:
> What would cause a ReceiveFrom on a pipe?

Receive calls ReceiveFrom here:

https://github.com/dotnet/runtime/blob/bcd07c52e2fcb698b405b3c59ba1c945601892b8/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs#L1508-L1512

> Similarly below, for SendTo

And Send calls SendTo here:

https://github.com/dotnet/runtime/blob/bcd07c52e2fcb698b405b3c59ba1c945601892b8/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs#L1845-L1849

Reply:
Yeah, I recall how this all works now, sorry.

I kind of wish that a simple socket Receive resulted in a simple call to recv(), instead of calling down into recvfrom() with a bunch of defaulted parameters.

But it probably doesn't matter in practice.